crafters/nlp/TikaExtractor/__init__.py

```diff
@@ -0,0 +1,83 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import os
+
+import requests
+from requests.exceptions import ConnectionError
+from jina.executors.crafters import BaseCrafter
+
+TIKA_URL = 'http://0.0.0.0:9998'
+
+
+class TikaExtractor(BaseCrafter):
+    """
+    :class:`TikaExtractor` Extracts text from files.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def post_init(self):
+        super().post_init()
+        self.start_tika()
+
+    def start_tika(self):
+        import sys
+        import time
+        from subprocess import Popen, PIPE
+
+        # Start tika server in subprocess
+        tika_version = os.getenv('TIKA_VERSION')
+        self.tika_process = Popen(
+            ['java', '-jar', f'/tika-server-{tika_version}.jar', '-h', '0.0.0.0'],
+            stdout=PIPE,
+            stderr=PIPE)
+
+        # Retry until tika server is started
+        for _ in range(10):
+            try:
+                time.sleep(1)
+                requests.get(TIKA_URL)
+                return
+            except ConnectionError:
+                pass
+        raise TimeoutError('Timeout when waiting for tika to start')
+
+    def close(self):
+        super().close()
+        self.tika_process.kill()
+
+    def craft(self, uri: str, buffer: bytes, *args, **kwargs):
+        from tika import parser
+        headers = {
+            'X-Tika-PDFOcrStrategy': os.getenv('TIKA_OCR_STRATEGY', 'ocr_only'),
```

Comment:
all these env variables, it is best if they are passed as parameters to the Executor I think

Reply:
Okay sure, do you have an example executor where we pass parameters?

Reply:
I mean in the constructor of the Executor, many executors take parameters with some default values